src/evict/evict_lru.c

```diff
@@ -1012,7 +1019,7 @@ retry:	while (slot < max_entries && ret == 0) {
 				cache->evict_file_next = NULL;
 			else
 				dhandle = TAILQ_FIRST(&conn->dhqh);
-		} else {
+		} else if (!F_ISSET(dhandle, WT_DHANDLE_DEAD) || dead_next) {
```

Comment:
Isn't that OK? Isn't a page marked `WT_READGEN_OLDEST` potentially more useful a page from a dead tree? (Unless `WT_READGEN_OLDEST` is code for "forcible eviction"?)